File: civitasRUBY/lib/sorpresa_por_casa_hotel.rb
# encoding:utf-8

module Civitas
  class SorpresaPorCasaHotel < Sorpresa
    
    attr_reader :texto, :valor
    
    def initialize(texto,_valor)
      super(texto)
      @valor=_valor      
    end
    
    def self.constructorSorpresaPorCasaHotel(texto,valor)
      new(texto,valor)
    end
    
    def aplicarAJugador(actual,todos)
      if(jugadorCorrecto(actual, todos))
        informe(actual, todos)
        todos[actual].modificarSaldo(@valor*(todos[actual].cantidadCasasHoteles))
      end
    end
    
    def informe(actual,todos)
      super(actual,todos)
    end
    
    public_class_method :new
    
  end
end

File: civitasJAVA/src/civitas/SorpresaPorCasaHotel.java
package civitas;

import java.util.ArrayList;

/**
 * @author Jesús Navarro Merino
 * @author José Alberto Gómez García
 */

public class SorpresaPorCasaHotel extends Sorpresa {

    private int valor;

    public SorpresaPorCasaHotel(String texto, int _valor) {
        super(texto);
        valor=_valor;
    }

    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
         if(jugadorCorrecto(actual,todos)){
            informe(actual,todos);
            (todos.get(actual)).modificarSaldo(valor*(todos.get(actual)).cantidadCasasHoteles()); //PERO ESTO SERIA PERDER DINERO NO?
        }
    }
    
    @Override
    protected void informe (int actual, ArrayList<Jugador> todos){
        super.informe(actual, todos);
    }
    
}
